lib/iris/fileformats/netcdf/loader.py

```diff
@@ -610,3 +657,132 @@ def load_cubes(file_sources, callback=None, constraints=None):
+
+        Notes
+        -----
+        This function acts as a contextmanager, for use in a 'with' block.
+
+        Example:
+
+        #todo
+        # >>> from iris.fileformats.netcdf.loader import CHUNK_CONTROL
+        # >>> from iris import sample_data_path
+        # >>> with CHUNK_CONTROL.set('var1', model_level=1, time=50):
+        # ...     cubes = iris.load(sample_data_path("toa_brightness_stereographic.nc"))
+
+        When ``var_names`` is present, the chunksize adjustments are applied
+        only to the selected variables.  However, for a CF data variable, this
+        extends to all components of the (raw) cube created from it.
+
+        **Un**-adjusted dimensions have chunk sizes set in the 'usual' way.
+        That is, according to the normal behaviour of
+        :meth:`iris._lazy_data.as_lazy_data`, which is : chunksize is based on
+        the file variable chunking, or full variable shape; this is scaled up
+        or down by integer factors to best match the Dask "default chunksize",
+        i.e. the setting configured by
+        ``dask.config.set({'array.chunk-size': '250MiB'})``.
+
+        """
+        old_settings = deepcopy(self.var_dim_chunksizes)
+        if var_names is None:
+            var_names = ["*"]
+        elif isinstance(var_names, str):
+            var_names = [var_names]
+        try:
+            for var_name in var_names:
+                # Note: here we simply treat '*' as another name.
+                # A specific name match should override a '*' setting, but
+                # that is implemented elsewhere.
+                if not isinstance(var_name, str):
+                    msg = (
+                        "'var_names' should be an iterable of strings, "
+                        f"not {var_names!r}."
+                    )
+                    raise ValueError(msg)
+                dim_chunks = self.var_dim_chunksizes.setdefault(var_name, {})
+                for dim_name, chunksize in dimension_chunksizes.items():
+                    if not (
+                        isinstance(dim_name, str)
+                        and isinstance(chunksize, int)
+                    ):
+                        msg = (
+                            "'dimension_chunksizes' kwargs should be an iterable "
+                            f"of (str, int) pairs, not {dimension_chunksizes!r}."
+                        )
+                        raise ValueError(msg)
+                    dim_chunks[dim_name] = chunksize
+            yield
+        finally:
+            self.var_dim_chunksizes = old_settings
+
+    @contextmanager
+    def from_file(self) -> None:
+        """
+        Ensures the chunks are loaded in from file variables, else will throw an error.
+
+        Notes
+        -----
+        This function acts as a contextmanager, for use in a 'with' block.
+        """
+        try:
+            self.mode = "file"
+            yield
+        finally:
+            self.mode = None
```

Comment:
Having read through everything, I think we might expected 'stacked' uses of this context manager in some cases. Would therefore be safer to:

- Store the previous state of `self.mode` and restore it afterwards
- Store the previous state of `var_dim_chunksizes` and restore it afterwards